### PR TITLE
Add body logging truncation

### PR DESCRIPTION
**Pull request by armstnp**

### CODE REVIEW

File: lib/smartsheet/api/request_logger.rb [hunk 1]
```diff
@@ -88,27 +91,34 @@ def log_api_error(level, response)
...
+        body_str =
+            if body.is_a? String
+              body
+            elsif body.is_a? Hash
+              PAYLOAD_CENSOR.censor_hash(body).to_s
+            else
+              '<Binary body>'
+            end
+
+        unless log_full_body || body_str.length <= TRUNCATED_BODY_LENGTH
```
Comment (cameronbowie) [inline lib/smartsheet/api/request_logger.rb]: Could you make this into a method?

File: test/smartsheet/api/request_logger_test.rb [hunk 2]
```diff
@@ -99,31 +106,50 @@ def given_mock_success_response(result: {}, headers: {})
...
+      given_mock_request(body: 'x' * TRUNCATION_LIMIT)
+      @request_logger.log_request(@mock_request)
+      @mock_logger.debug_msgs.must_include "Request Body: #{'x' * TRUNCATION_LIMIT}"
+    end
+
+    it 'does not truncate bodies when the `log full body` flag is true' do
+      given_full_body_flag
+      given_mock_request(body: 'x' * OVER_TRUNCATION_LIMIT)
+      @request_logger.log_request(@mock_request)
+      @mock_logger.debug_msgs.must_include "Request Body: #{'x' * OVER_TRUNCATION_LIMIT}"
```
Comment (cameronbowie) [inline test/smartsheet/api/request_logger_test.rb]: Could you add a test to verify requests with Hash bodies are truncated?

Comment (armstnp) [reply test/smartsheet/api/request_logger_test.rb]: Good call.

File: lib/smartsheet/api/request_logger.rb (hunk 1 above)
Comment (armstnp) [reply lib/smartsheet/api/request_logger.rb]: :+1:

**Pull request merged**